### ISSUE TITLE
The column 'w' on table 'oc_memories' already exists.

### ISSUE THREAD

**Issue by karpiq24** [pulsejet/memories#164]
I cannot install and enable memories beacuse of `The column 'w' on table 'oc_memories' already exists.`

```
[settings] Error: Doctrine\DBAL\Schema\SchemaException: The column 'w' on table 'oc_memories' already exists. at <<closure>>

 0. /config/www/nextcloud/3rdparty/doctrine/dbal/src/Schema/Table.php line 473
    Doctrine\DBAL\Schema\SchemaException::columnAlreadyExists()
 1. /config/www/nextcloud/3rdparty/doctrine/dbal/src/Schema/Table.php line 355
    Doctrine\DBAL\Schema\Table->_addColumn()
 2. /config/www/nextcloud/apps/memories/lib/Migration/Version400000Date20221015121115.php line 58
    Doctrine\DBAL\Schema\Table->addColumn()
 3. /config/www/nextcloud/lib/private/DB/MigrationService.php line 445
    OCA\Memories\Migration\Version400000Date20221015121115->changeSchema()
 4. /config/www/nextcloud/lib/private/DB/MigrationService.php line 408
    OC\DB\MigrationService->migrateSchemaOnly()
 5. /config/www/nextcloud/lib/private/Installer.php line 153
    OC\DB\MigrationService->migrate()
 6. /config/www/nextcloud/apps/settings/lib/Controller/AppSettingsController.php line 451
    OC\Installer->installApp()
 7. /config/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 225
    OCA\Settings\Controller\AppSettingsController->enableApps()
 8. /config/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 133
    OC\AppFramework\Http\Dispatcher->executeController()
 9. /config/www/nextcloud/lib/private/AppFramework/App.php line 172
    OC\AppFramework\Http\Dispatcher->dispatch()
10. /config/www/nextcloud/lib/private/Route/Router.php line 298
    OC\AppFramework\App::main()
11. /config/www/nextcloud/lib/base.php line 1047
    OC\Route\Router->match()
12. /config/www/nextcloud/index.php line 36
    OC::handleRequest()

POST /settings/apps/enable
```

The app shows as installed but I can't enable it beacuse of this error. Removing and reinstalling doesn't fix this issue.

**Comment by karpiq24** [comment]
I fixed it by removing memories app and manually deleting memories database table. After that I tried to install this app again and it worked. I don't know what caused the initial problem. Issue van be closed if it's not worth investigating.

**Comment by pulsejet** [comment]
Your database probably entered some invalid state because of an incomplete migrations. It can happen sometimes.

**Comment by ralfi** [comment]
> I fixed it by removing memories app and manually deleting memories database table. After that I tried to install this app again and it worked this time. I don't know what caused the initial problem. Issue van be closed if it's not worth investigating.

Hi, same error here. How do you delete the memories database table? Thanks a lot.

**Comment by pulsejet** [comment]
```sql
DROP TABLE `oc_memories`;
```

**Comment by ralfi** [comment]
Yeah, make sense... ;-)

**Comment by farhills** [comment]
I'm having this issue as well.  dropped 'oc_memories' and removed the app.

After install, the app shows installed, but not enabled.  Click 'enable' and the column 'w' error appears.

![image](https://user-images.githubusercontent.com/83388006/208177781-c95f5769-191e-4a64-9cc9-1f4e585fe116.png)


This is from the server log:

[settings] Error: Doctrine\DBAL\Schema\SchemaException: The column 'w' on table 'oc_memories' already exists. at <<closure>>

 0. /config/www/nextcloud/3rdparty/doctrine/dbal/src/Schema/Table.php line 473
    Doctrine\DBAL\Schema\SchemaException::columnAlreadyExists()
 1. /config/www/nextcloud/3rdparty/doctrine/dbal/src/Schema/Table.php line 355
    Doctrine\DBAL\Schema\Table->_addColumn()
 2. /config/www/nextcloud/apps/memories/lib/Migration/Version400000Date20221015121115.php line 58
    Doctrine\DBAL\Schema\Table->addColumn()
 3. /config/www/nextcloud/lib/private/DB/MigrationService.php line 445
    OCA\Memories\Migration\Version400000Date20221015121115->changeSchema()
 4. /config/www/nextcloud/lib/private/DB/MigrationService.php line 408
    OC\DB\MigrationService->migrateSchemaOnly()
 5. /config/www/nextcloud/lib/private/Installer.php line 153
    OC\DB\MigrationService->migrate()
 6. /config/www/nextcloud/apps/settings/lib/Controller/AppSettingsController.php line 451
    OC\Installer->installApp()
 7. /config/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 225
    OCA\Settings\Controller\AppSettingsController->enableApps()
 8. /config/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 133
    OC\AppFramework\Http\Dispatcher->executeController()
 9. /config/www/nextcloud/lib/private/AppFramework/App.php line 172
    OC\AppFramework\Http\Dispatcher->dispatch()
10. /config/www/nextcloud/lib/private/Route/Router.php line 298
    OC\AppFramework\App::main()
11. /config/www/nextcloud/lib/base.php line 1047
    OC\Route\Router->match()
12. /config/www/nextcloud/index.php line 36
    OC::handleRequest()

POST /settings/apps/enable
from x.x.x.x by Farhills at 2022-12-16T19:48:19+00:00

**Comment by pulsejet** [comment]
@farhills did you clean up everything first? See https://github.com/pulsejet/memories/issues/295

**Comment by farhills** [comment]
Looks like memories is bagholding for other issues on my server. After the failed install I discovered the Dav interface was down. Still working the issues, but looks like an app didn't update cleanly earlier in the day but things didn't blow up until I tried to install memories.

I'll uninstall memories, drop the table and tidy up the other DB loose ends. Will report back with the steps that (hopefully) worked. Thanks!

**Comment by farhills** [comment]
Update: Cleaned up the DB (mariadb) per instructions above and #295.

Removed Memories via 'apps' in the NC web interface, then in MariaDB:

DROP TABLE IF EXISTS oc_memories, oc_memories_livephoto;

DELETE FROM oc_migrations WHERE app = 'memories';

DROP INDEX memories_parent_mimetype ON oc_filecache; 

This failed on the first try.  Not exactly sure why, my suspicion is Redis or another cache was remembering some of the deleted entries.  On the second attempt I restarted Redis, Mariadb and Nextcloud together before attempting to install memories.  This time it worked!

Thanks for pointing me to #294, I'd have been stuck without it.
